proposals/custom-feeds.md

```diff
@@ -0,0 +1,188 @@
+
+- `Unreachable` - No server was reached.
+- `ServerError` - No server responded a status code `200`.
+  (Caveat: Redirects should be followed.)
+- `Unparseable` - A server responded with `200`, but the data did not match the form the definition.
+- `PhaseOneValidationFail` - Fails Phase One validation.
+
+##### Validation & Aggregation definition
+
+Validator definition parts:
+
+1. Rerun phase one validation.
+   Note that for time bounds this is relative to the collector's claimed time of collection.
+   Any failure will result in collector not being part of the aggregation.
+2. Phase two validation/ aggregation:
+   This may involve logic applied to multiple collector results.
+
+Phase two may fail in some cases if, say,
+the divergence is too great, or there are too many reported errors,
+or there is insufficient successful collector results.
+
+The output is:
+
+- the fact statement to be published on Cardano
+- the "data trace" to be put on Arweave
+
+##### Publish
+
+A fact statement is of the form:
+
+```ts
+{
+    id: X/<name>/<version>,
+    created_at: Timestamp,
+    body: Data,
+}
+```
+
+This is the same as is currently in use.
+Consumer data will be delivered inside `body`.
+
+The feed id `version` is owner's key (raw bytes, **NOT** hex encoded).
+It is up to the customer to communicate to downstream consumers
+as to whether or not they ought to be verifying the full feed id (including version).
+
+Orcfax must lay out some examples of feed definitions for customers to understand this stage.
+
+### On-chain registration
+
+The customer performs on-chain registration.
+This is a transaction that creates an OrcfaxRegister token at the corresponding address
+with an inline datum with the following data:
+
+```ts
+{
+    owner: VerificationKey,
+    info: {
+        hot-key: VerificationKey (Scheme dependent),
+        source: [<HTTP_ENDPOINT>],
+        offline-registration-url: <URL>,
+        repo-version: <GIT_HASH>,
+    }
+}
+```
+
+- A token is minted, with name equal to customers cold key
+- Future (continuing) spends, (or end) must be singed by the owner key
+- The hot key can be updated.
+- Source info is probably : URL(s) and the expect request info (HTTP/JSON/BYTES).
+
+The feed id will use the customer's cold key digest as the version.
+
+- Other members of the validator network can independently verify the signatures
+  and do not need to collect the data separately.
+
+### Open questions
+
+There are many open questions concerning how rigid or flexible this process ought to be.
+Here are just some.
+
+1. Data source schemes: do we support json and bytes
+
+- Plutus data encoded.
+- Data as bytestring
+
+2. Phase one validation: do we support more than signatures.
+   If so, we need to parse the bodies and extract, say, signatures.
+   Which schemes and formats do we support?
+
+3. What if there are multiple on-chain registrations of the same owner?
+
+4. What precisely is the point of the on-chain registration?
```

Comment:
Decentralization for collectors is one.

5. what exactly does de-registration look like? 
6. does orcfax have the power to deregister based on solid rules? e.g. endpoints are down for x-hours?